src/shared/psi-util.c

```diff
@@ -107,12 +108,20 @@ int read_resource_pressure(const char *path, PressureType type, ResourcePressure
 int is_pressure_supported(void) {
         const char *p;
 
-        FOREACH_STRING(p, "/proc/pressure/cpu", "/proc/pressure/io", "/proc/pressure/memory")
-                if (access(p, F_OK) < 0) {
-                        if (errno == ENOENT)
-                                return 0;
-                        return -errno;
-                }
+        /* The pressure files, both under /proc and in cgroups, will exist
+         * even if the kernel has PSI support disabled; we have to read
+         * the file to make sure it doesn't return -EOPNOTSUPP */
+        FOREACH_STRING(p, "/proc/pressure/cpu", "/proc/pressure/io", "/proc/pressure/memory") {
+                _cleanup_free_ char *b = NULL;
+                size_t s;
+                int r;
+
+                r = read_virtual_file(p, 1, &b, &s);
```

Comment:
you can pass the fourth argument as NULL, given you are not actually interested in the result here.

And I think we should also make it possible to pass the third argument as NULL given that we are not interested in the result for that either. I prepped a PR for that in #19680. Maybe pull that commit into this PR, and then rebase this on top passing both arguments as NULL?